packages/default-reporter/src/reporterForClient/reportProgress.ts

```diff
@@ -156,16 +172,41 @@ function getProgessStatsPushStreamByRequirer (progress$: most.Stream<ProgressLog
   return progessStatsPushStreamByRequirer
 }
 
+function getPackageImportMethodPushStream (packageImportMethod$: most.Stream<PackageImportMethodLog>) {
+  const packageImportMethodPushStream = new PushStream<string>()
+
+  let packageImportMethod = 'hard linked'
+  packageImportMethod$
+    .forEach((log: PackageImportMethodLog) => {
+      switch (log.method) {
+        case 'copy':
+          packageImportMethod = 'copied'
+          break
+        case 'clone':
+          packageImportMethod = 'cloned'
+          break
+        default:
+          break
+      }
+      packageImportMethodPushStream.next(packageImportMethod)
+    })
+
+  return packageImportMethodPushStream
+}
+
 function createStatusMessage (
+  showImportingMethod: boolean,
   progress: ProgressStats,
+  importingMethod: string,
   importingDone: boolean
 ) {
   const msg = `Resolving: total ${hlValue(progress.resolved.toString())}, reused ${hlValue(progress.reused.toString())}, downloaded ${hlValue(progress.fetched.toString())}`
   if (importingDone) {
+    const importingMsg = `Packages were ${importingMethod} from the content-addressable store to the virtual store.\nContent-addressable store is at: ~/.pnpm-store/v3\nVirtual store is at: node_modules/.pnpm`
     return {
       done: true,
       fixed: false,
-      msg: `${msg}, done`,
+      msg: `${msg}, done${showImportingMethod ? '\n' + importingMsg : ''}`,
```

Comment:
I think this should be a new type of reporter. There's no reason to glue it to the progress reporter. Just make a new one, like `reportContext.ts`